ci/taos/plugins-good/pr-format-misspelling.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+##
+# Copyright (c) 2018 Samsung Electronics Co., Ltd. All Rights Reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##
+# @file     pr-format-misspelling.sh
+# @brief    Check a misspelled statement in a text document file with GNU Aspell
+#
+# GNU Aspell is a Free and Open Source spell checker designed to eventually replace Ispell.
+# It can either be used as a library or as an independent spell checker. Its main feature
+# is that it does a superior job of suggesting possible replacements for a misspelled word
+# than just about any other spell checker out there for the English language. Unlike Ispell,
+# Aspell can also easily check documents in UTF-8 without having to use a special dictionary.
+#
+# @see      http://aspell.net/
+# @see      https://github.com/nnsuite/TAOS-CI
+# @author   Geunsik Lim <geunsik.lim@samsung.com>
+#
+
+# @brief [MODULE] TAOS/pr-format-misspelling
+function pr-format-misspelling(){
+echo "########################################################################################"
+echo "[MODULE] TAOS/pr-format-misspelling: Check a misspelled statement in a document file with GNU Aspell"
+
+# Check if required commands are installed by server administrator
+check_dependency wc
+check_dependency cat
+check_dependency aspell
+
+# Read file names that a contributor modified(e.g., added, moved, deleted, and updated) from a last commit.
+FILELIST=`git show --pretty="format:" --name-only --diff-filter=AMRC`
+
+# Inspect all files that contributor modifed.
+for i in ${FILELIST}; do
+    # skip obsolete folder
+    if [[ $i =~ ^obsolete/.* ]]; then
+        continue
+    fi
+    # skip external folder
+    if [[ $i =~ ^external/.* ]]; then
+        continue
+    fi
+    # Handle only text files in case that there are lots of files in one commit.
+    echo "[DEBUG] file name is ( $i )."
+    if [[ `file $i | grep "ASCII text" | wc -l` -gt 0 ]]; then
+        # in case of document file: *.md, *.txt)
+        case $i in
+            # in case of MarkDown(MD) and text file
+            *.md | *.txt)
+                echo "[DEBUG] ( $i ) file is a document file with a ASCII text format."
+                typo_analysis_sw="aspell"
+                typo_analysis_rules=" list -l en "
+                typo_check_result="misspelling_result.txt"
+                fi
+                $typo_analysis_sw $typo_analysis_rules  > ../report/${typo_check_result}
```

Comment:
How can aspell check '$i' file?

Reply:
Nice catch. Fixed. 
[FYI] (potential misspelling characters are listed in  `../report/${typo_check_result}` file.